runtime/onert/core/src/odc/CodegenManager.cc

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright (c) 2024 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "CodegenLoader.h"
+#include "odc/CodegenManager.h"
+#include "util/Utils.h"
+
+#include <fstream>
+#include <mutex>
+
+namespace onert
+{
+namespace odc
+{
+
+bool CodegenManager::codegen(const char *target, CodegenPreference pref)
+{
+  // codegen function is thread-unsafe
+  static std::mutex lock;
+  std::lock_guard<std::mutex> guard(lock);
+
+  if (_export_model_path.empty())
+    throw std::runtime_error("Export model path is not set");
+
+  std::ifstream file(_model_path);
+  if (!file.good())
+    throw std::runtime_error("Model path does not exist");
+
+  auto &codegen_loader = CodegenLoader::instance();
```

Comment:
I think here may cause thread issue. Am I understanding right?

Reply:
If so, we can move down mutex code to the above this code.

Reply:
(Out of curiosity)
What thread issue does here cause?

Reply:
I see. You mean that this function may be thread-safe without using `std::mutex` in this function if we make `CodegenLoader::instance()` thread-safe?t

Reply:
👍